qc-runner/src/dataIntegration.r

```diff
@@ -109,13 +111,31 @@ run_dataIntegration <- function(scdata, config){
             data.split[[i]] <- Seurat::NormalizeData(data.split[[i]], normalization.method = normalization, verbose = F)
             data.split[[i]] <- Seurat::FindVariableFeatures(data.split[[i]], selection.method = "vst", nfeatures = nfeatures, verbose = FALSE)
         }
-        data.anchors <- Seurat::FindIntegrationAnchors(object.list = data.split, dims = 1:numPCs, verbose = FALSE)
+        # If Number of anchor cells is less than k.filter/2, there is likely to be an error:
+        # Note that this is a heuristic and was found to still fail for small data-sets
+
+        # Try to integrate data (catch error most likely caused by too few cells)
+
+        tryCatch({
+          k.filter <- min(ceiling(sapply(data.split, ncol)/2), k.filter)
+          data.anchors <- Seurat::FindIntegrationAnchors(object.list = data.split, dims = 1:numPCs, k.filter = k.filter, verbose = TRUE)
```

Comment:
```suggestion
          data.anchors <- Seurat::FindIntegrationAnchors(object.list = data.split, dims = 1:numPCs, k.filter = k.filter, verbose = FALSE)
```
Do we want to leave verbosity on?